tools/js2c.cc

```diff
@@ -902,9 +909,9 @@ int Main(int argc, char* argv[]) {
     } else if (error != 0) {
       return 1;
     } else {  // It's a file.
-      std::string_view extension = HasAllowedExtensions(file);
-      if (extension.size() != 0) {
-        auto it = file_map.insert({std::string(extension), FileList()}).first;
+      auto extension = HasAllowedExtensions(file);
+      if (extension.has_value()) {
```

Comment:
Because node uses C++17, you could do

```C++
if (auto extension = HasAllowedExtensions(file); extension.has_value()) {
```

(This is *NOT* a request for change, just a comment.)